x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/GetDeploymentStatsAction.java

```diff
@@ -293,6 +301,9 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
                 if (reason != null) {
                     builder.field("reason", reason);
                 }
+                if (health != null) {
+                    builder.field("health", health);
```

Comment:
I would make `health` an object field, the existing `AllocationHealth` enum can be one of the fields of this object.  Other fields can be `allocation_count` and `target_allocation_count`.

I think in the future we would regret making this field a scalar, because it's more disruptive and harder to achieve BWC when converting a scalar to an object than it is when adding or removing fields from an existing object.

And as we move away from the MVP of "allocate models to every node" we'll want to wait for more complex states than simply "on one node" and "on every node".

Also in the future for debugging CPU-based autoscaling it will be interesting to see what the stats consider the target allocation count to be.  So I am pretty sure we'll end up wanting these fields in the future even though they seem redundant now.

Reply:
'health` is no longer an appropriate name for this field when the enum values are ` starting`, `started` & `fully_allocated`  and with the new count fields. Should we use the old reliable `status`?

Reply:
I don't see how `health` isn't still a valid option. 

Now we have allocation state vs allocation status, its confusing. I would like a more differentiable option than `status` as its only 2 characters away from `state`.

Reply:
One way around this might be:

1. Call the top level field `allocation`
2. Have `health`, `count`, `target_count` and `nodes` as sub-fields of this (with `nodes` moving down a level from where it is now)

Reply:
@davidkyle I do think that maybe `allocation_status` might be ok. Instead of simply `status`.